include/autoppl/expression/traits.hpp

```diff
@@ -8,12 +9,14 @@ namespace ppl {
  * Users should rely on these classes to grab member aliases.
  */
 
-template <class TagType>
-struct tag_traits
+template <class VarType>
+struct var_traits
 {
-    using value_t = typename TagType::value_t;
-    using pointer_t = typename TagType::pointer_t;
-    using state_t = typename TagType::state_t;
+    using value_t = typename VarType::value_t;
+    using pointer_t = typename VarType::pointer_t;
+    using state_t = typename VarType::state_t;
+
+    static_assert(std::is_convertible_v<VarType, value_t>);
```

Comment:
We can keep this here for now, but I think we'll have to move it later. Can you add a TODO like "this may have to move in a different class for compile-time checking". Def not a concern at this pt tho.